be/src/storage/rowset/segment_writer.cpp

```diff
@@ -146,6 +146,10 @@ Status SegmentWriter::init(const std::vector<uint32_t>& column_indexes, bool has
         // TODO(mofei) refactor it to type specification
         opts.need_zone_map = column.is_key() ||
                              (_tablet_schema->keys_type() == KeysType::DUP_KEYS && is_zone_map_key_type(column.type()));
+        if (column.use_zone_map() && (_tablet_schema->keys_type() == KeysType::DUP_KEYS ||
+                                      _tablet_schema->keys_type() == KeysType::PRIMARY_KEYS)) {
+            opts.need_zone_map = true;
```

Comment:
Do we need an guard check to make sure data type is supported for the `use_zone_map` columns?

Reply:
Check when create table and alter table.